flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamContextEnvironment.java

```diff
@@ -75,29 +62,31 @@ public JobExecutionResult execute() throws Exception {
 	@Override
 	public JobExecutionResult execute(String jobName) throws Exception {
 
-		JobGraph jobGraph;
-		if (jobName == null) {
-			jobGraph = this.getStreamGraph().getJobGraph();
-		} else {
-			jobGraph = this.getStreamGraph().getJobGraph(jobName);
-		}
-
-		transformations.clear();
-
-		// attach all necessary jar files to the JobGraph
-		for (URL file : jars) {
```

Comment:
The detached job goes through `Client#getJobGraph(FlinkPlan, List<URL>, List<URL>)` where the jars and classpaths are added to the job graph.